node/flatpak_node_generator/providers/npm.py

```diff
@@ -134,6 +134,9 @@ def _process_packages_v2(
                         'Git sources in lockfile v2 format are not supported yet'
                         f' (package {install_path} in {lockfile})'
                     )
+            elif info.get('optional'):
+                # Optional dependencies can be skipped
```

Comment:
Should this at least be printing out a warning? Not sure if we do that anywhere else in the source yet, but it could be useful if the user was actually expecting the optional dep to still make its way inside.